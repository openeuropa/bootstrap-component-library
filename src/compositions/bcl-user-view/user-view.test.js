import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-user-view/dataUserView";
import demoDataCompact from "@openeuropa/bcl-user-view/dataUserViewCompact";

const template = "@oe-bcl/bcl-user-view/user-view.html.twig";
const templateCompact = "@oe-bcl/bcl-user-view/user-view-compact.html.twig";

const render = (params) => renderTwigFileAsNode(template, params);
const renderCompact = (params) => renderTwigFileAsNode(templateCompact, params);

describe("OE - User view", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});

describe("OE - User view compact", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(renderCompact(demoDataCompact)).resolves.toMatchSnapshot();
  });
});
