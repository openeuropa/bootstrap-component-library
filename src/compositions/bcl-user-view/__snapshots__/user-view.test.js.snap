// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - User view compact renders correctly 1`] = `
<jest>
  <div
    class="d-flex mb-3"
  >
    <div
      class="flex-shrink-0"
    >
      <img
        alt="Stefan Mayer profile picture"
        class="rounded-pill"
        src="https://picsum.photos/50/50"
      />
    </div>
    <div
      class="flex-grow-1 ms-3"
    >
      <a
        class="fs-5 text-underline-hover mb-0"
        href="/example.html"
      >
        Stefan Mayer
      </a>
      <div
        class="d-flex flex-wrap"
      >
        <span
          class="text-muted text-nowrap me-4-5"
        >
          DG Test
        </span>
        <span
          class="text-muted text-nowrap"
        >
          Director
        </span>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - User view renders correctly 1`] = `
<jest>
  <main>
    <div
      class="container mt-5"
    >
      <div
        class="row"
      >
        <div
          class="col-12 col-md-10"
        >
          <div
            class="border-0 mb-md-3 card"
          >
            <div
              class="row"
            >
              <div
                class="col-6 col-md-3 offset-3 offset-md-0"
              >
                <img
                  alt="alt img"
                  class="card-img-top"
                  src="https://picsum.photos/200/200?random=6"
                />
              </div>
              <div
                class="col-md-9"
              >
                <div
                  class="card-body pt-md-0 px-0 px-md-3"
                >
                  <h2
                    class="card-title fw-bold"
                  >
                    Stefan Mayer
                  </h2>
                  <div
                    class="mb-4 mt-4"
                  >
                    <span
                      class="text-muted me-3"
                    >
                      DG Test
                    </span>
                    <span
                      class="text-muted me-3"
                    >
                      Director
                    </span>
                  </div>
                  <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse volutpat ultricies massa, a dapibus libero porta nec. Sed facilisis dictum vestibulum. Fusce commodo hendrerit diam, pretium tempus leo varius sit amet. Etiam interdum, orci at sagittis luctus, lorem libero tempus mauris, a fermentum libero orci semper lacus. Duis tristique fringilla magna, eu egestas dolor molestie non. 
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-2 text-end"
        >
          <div
            class="d-grid d-md-block"
          >
            <button
              class="mb-3 btn btn-secondary btn-md"
              type="button"
            >
              Manage profile
            </button>
          </div>
        </div>
      </div>
      <div
        class="my-3"
      >
        <h4
          class="fw-bold mb-4"
        >
          Personal information
        </h4>
        <div
          class="row"
        >
          <div
            class="col-12 col-md-6"
          >
            <div
              class="row"
            >
              <div
                class="col-12 col-md-4"
              >
                <p
                  class="mb-0"
                >
                  Gender:
                </p>
              </div>
              <div
                class="col-12 col-md-8"
              >
                <p>
                  Male
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-12 col-md-4"
              >
                <p
                  class="mb-0"
                >
                  Nationality:
                </p>
              </div>
              <div
                class="col-12 col-md-8"
              >
                <p>
                  German
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-12 col-md-4"
              >
                <p
                  class="mb-0"
                >
                  Date of birth:
                </p>
              </div>
              <div
                class="col-12 col-md-8"
              >
                <p>
                  22/01/1972
                </p>
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-6"
          >
            <div
              class="row"
            >
              <div
                class="col-12 col-md-4"
              >
                <p
                  class="mb-0"
                >
                  Country:
                </p>
              </div>
              <div
                class="col-12 col-md-8"
              >
                <p>
                  Germany
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-12 col-md-4"
              >
                <p
                  class="mb-0"
                >
                  City:
                </p>
              </div>
              <div
                class="col-12 col-md-8"
              >
                <p>
                  DÃ¼sseldorf
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="my-3"
      >
        <h4
          class="fw-bold mb-4"
        >
          Professional information
        </h4>
        <div
          class="row"
        >
          <div
            class="col-12 col-md-6"
          >
            <div
              class="row"
            >
              <div
                class="col-12 col-md-4"
              >
                <p
                  class="mb-0"
                >
                  Organisation:
                </p>
              </div>
              <div
                class="col-12 col-md-8"
              >
                <p>
                  DG Test
                </p>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-12 col-md-4"
              >
                <p
                  class="mb-0"
                >
                  Position:
                </p>
              </div>
              <div
                class="col-12 col-md-8"
              >
                <p>
                  Director
                </p>
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-6"
          >
            <div
              class="row"
            >
              <div
                class="col-12 col-md-4"
              >
                <p
                  class="mb-0"
                >
                  Working languages:
                </p>
              </div>
              <div
                class="col-12 col-md-8"
              >
                <p>
                  English, German
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</jest>
`;
